src/excmds.ts

```diff
@@ -1706,11 +1706,65 @@ export async function tabprev(increment = 1) {
     // Proper way:
     // return tabIndexSetActive((await activeTab()).index - increment + 1)
     // Kludge until https://bugzilla.mozilla.org/show_bug.cgi?id=1504775 is fixed:
-    return browser.tabs.query({ currentWindow: true, hidden: false }).then(tabs => {
-        tabs = tabs.sort((t1, t2) => t1.index - t2.index)
-        let prevTab = (tabs.findIndex(t => t.active) - increment + tabs.length) % tabs.length
-        return browser.tabs.update(tabs[prevTab].id, { active: true })
-    })
+
+    if (config.get("treestyletabintegration")) {
+        // Ok so this entire piece here is really inefficient, it looks up all tabs, gets the active tab id, gets all tabs again (this time as a tree), iterates through all those tabs flattening them into an array, and then iterates over them once more to find the index of the active tab in that flattened array. This has a lot of room for improvement in the future.
+        return browser.tabs.query({ currentWindow: true, hidden: false }).then(tabs => {
+            tabs = tabs.sort((t1, t2) => t1.index - t2.index)
+            // Find the current TAB id
+            let activeTabId = tabs[tabs.findIndex(t => t.active)].id
```

Comment:
You could use  `activeTabId()` here instead of doing this. The definition lives in `src/lib/webext.ts` It has some comon shortcuts for things you're likely to do a lot when messing with excmds.

Reply:
Fixed it so it uses activeTabId, and will be throwing it in it's own file.